docs/concept-workflows.md

```diff
@@ -1,30 +1,209 @@
+:::note
 
-Here is an example Workflow that implements the subscription management use case. It is in Java, but Go is also supported. [Other SDKS](/docs/sdks-introduction/#other-sdks) are under active development.
+The [Event Sourcing pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/event-sourcing) article published in Microsoft's Azure documentation provides a general explanation of what that technique looks like.
+
+:::
+
+As a result, a Workflow must use certain Temporal SDK APIs to manage time related operations or create new threads.
+And it also means that any "non-deterministic" behavior, such as calls to external APIs, or random number generators, must not exist directly inside the Workflow.
+The elegant solution to this is to have any code that handles this potentially unexpected behavior reside in [Activities](/docs/concept-activities).
+
+But, by using event sourcing a Workflow becomes quite durable and in essence "fault-tolerant".
+For example, if a host fails in the middle of a Workflow execution, not only can another host resume the Workflow, but it can resume at the exact line of code where it stopped executing without having to re-execute the code up until that point.
+The Workflow execution state is even preserved through crashes of the Temporal Server itself.
+
+## Freedom to relocate
+
+While there are a few specific use-cases that may require a Workflow to start and finish on the same host, Workflow code is completely independent from the location in which it is executed
+
+Instead of deploying executable code to a host directly, Temporal applications use special [Worker](/docs/concept-workers) processes whose sole job is to host and execute Workflow and Activity code.
+The Worker communicates with the Temporal Server in order to know which line of code to execute through [Tasks](/docs/concept-task-queues) it picks up.
+
+Because a Workflow is executed piece by piece, and because the Server tracks the state of that execution, there can be multiple Worker processes in multiple locations capable of executing any one of those pieces.
+
+Essentially load balancing and processing failover are "out-of-the-box" features of Temporal Workflows.
+
+## Workflows have options
+
+When you start a Workflow, you can pass along parameters that tell the Temporal Server how to handle the Workflow.
+This includes the ability to set timeouts for Workflow execution, a retry policy, the Task Queue name, a data converter, and child Workflow options.
+
+### Timeout settings
+
+It's often necessary to limit the amount of time a specific Workflow can be running.
+
+To support this, the following three parameters can be provided to Workflow options:
+
+#### Execution timeout
+
+- **Description**: This is the maximum amount of time that a Workflow should be allowed to run including retries and any usage of the "Continue-as-new" feature.
+This is different from [Run timeout](#run-timeout).
+- **Use-case**:
```

Comment:
@mfateev - I added specific use-case callouts to the Workflow timeout and retry info. Any way to capture these in a few sentences?